[MRG] Opt-in sharing, symptom reports improvements

madprime opened the pull request:
This adds opt-in sharing for retrospective reports ("publish") and ongoing symptom tracking.

Maybe ready to merge – but needs review, especially of the text and UX.

gedankenstuecke commented on reports/templates/reports/list.html:
```diff
@@ -5,13 +5,60 @@
   {% endif %}
 </h2>
 
+{{ request.user.openhumansmember.account.publish_symptom_reports }}
```
This prints true/false to the template and can be removed :-)

madprime commented:
> I thought from the call we would go with "you will see that option once you schedule email reminders" but that's not the case? Right now it's either on /manage_account/ or you see if it triggering a symptom report through the web interface? (Maybe also on filing a no symptom report via email?)

Yeah, I felt like the "check-in scheduling" wasn't the right place, or at least not that place alone. When scheduling check-ins, someone probably won't have a good sense of what it is they'll be sharing – what that data will look like.

Any symptom report (or "no symptom" report) should land the project member on their list of symptom reports, where the prompt for sharing is. This way, they can decide at any time (e.g. after having logged for a week). I think this is more consistent with the approach for retrospective events: prompt & decide to share when seeing the data that will be shared.

> The /manage_account/ still allows to set a public data flag, but to my understanding this flag now doesn't do anything anymore, as the sharing is now handled a) on a per-retrospective-data analysis basis and b) on a symptom-report wide level?

I guess this would imply language changes will need to occur elsewhere, right? There's no "opt-out" for any aggregate de-identified data, we just won't do that? Or maybe we will do so if it makes sense, and we don't think it's necessary to provide an opt-out? (The latter makes sense to me… maybe someday there will be data where it makes sense to share some sort of aggregate.)

gedankenstuecke commented:
> Yeah, I felt like the "check-in scheduling" wasn't the right place, or at least not that place alone. When scheduling check-ins, someone probably won't have a good sense of what it is they'll be sharing – what that data will look like.
> 
> Any symptom report (or "no symptom" report) should land the project member on their list of symptom reports, where the prompt for sharing is. This way, they can decide at any time (e.g. after having logged for a week). I think this is more consistent with the approach for retrospective events: prompt & decide to share when seeing the data that will be shared.

Okay, that's fair! 

>>  The /manage_account/ still allows to set a public data flag, but to my understanding this flag now doesn't do anything anymore, as the sharing is now handled a) on a per-retrospective-data analysis basis and b) on a symptom-report wide level?
>
>I guess this would imply language changes will need to occur elsewhere, right? There's no "opt-out" for any aggregate de-identified data, we just won't do that? Or maybe we will do so if it makes sense, and we don't think it's necessary to provide an opt-out? (The latter makes sense to me… maybe someday there will be data where it makes sense to share some sort of aggregate.)

I guess I have a hard time seeing how the de-identified & aggregated data sharing would look any different from the way we're exposing shared data already? Also in those cases it would give a user-ID to associate multiple data points along with all the data recordings? I think my point is that it might be confusing for people if there's a default data sharing in aggregate while at the same time giving them the option to not make data bits public?

madprime commented:
Agreed it's confusing, agree the best thing to do is remove it.

I'm thinking about what might happen if one-off surveys were added. I think it's unclear how that'll look or what might be done with it, but: associating those with other data is likely to make data more identifiable.

But maybe what people agree to for that data should (as with these data!) occur if/when such surveys occur. Sound right?

gedankenstuecke commented:
Yep, sounds good!

madprime commented:
I wanted to tie up more loose ends, even though a lot are minor…
* updated the form to have a slider
* minor language revision on symptoms (based on feedback from Gary)
* added an upload of the symptom report data into Open Humans
* added links for getting public symptom reporting data & associated text

Ideally the public symptom reporting data would include wearable data that might be visualized alongside it, but this is what we have right now.

See what you think! I hope this all works fine, but you caught a bug before. Please go ahead and merge if you think it's good. :)

gedankenstuecke commented on reports/models.py:
```diff
@@ -140,18 +179,29 @@ def severity(self):
         return 4
 
     def as_json(self):
+        if isinstance(self.fever, type(None)):
```
I added this check here as the sync to Open Humans would otherwise fail for me, saying that type `Decimal` can't be cast into JSON. This way it's either a proper number or an empty string at this point.

gedankenstuecke commented on quantified_flu/templates/quantified_flu/home.html:
```diff
@@ -22,8 +22,8 @@ <h3>Track symptoms going forward</h3>
 <p>
-  We will <b>share the aggregated and deidentified data</b> with researchers and publicly, to maximize the impact of your data!
-  This includes some weeks of wearable data around the sick dates, as well as the reported symptoms. But of course you can always opt-out of this!
+  You can opt-in to <b>share the aggregated and deidentified data</b> with researchers and publicly, to maximize the impact of your data!
```
I rewrote this part to be in line with the `/about` text and how the website actually works.

gedankenstuecke commented on reports/templates/reports/list.html:
```diff
@@ -41,8 +75,8 @@ <h2>
+          {{ symptomitem.symptom.get_label_display }} ({{ symptomitem.get_intensity_display }}){% if not forloop.last %}, {% endif %}
           {% endfor %}
         {% endif %}
       </td>
```
I can't comment on the lines otherwise as I missed it, but I noticed that this table is never showing the actual fever temperature recording in it. Would maybe be good to add that too?
